
describe('retryability', () => {
  // this test will fail on first try, but pass on second
  it('Retry resets api state', { retries: 1 }, () => {

    // @ts-ignore cy.state() has no type definitions
    const currentRetry = cy.state('runnable')._currentRetry

    // making sure we donâ€™t see requests from previous retry
    cy.get('[data-cy="responseBody"]').should('not.exist')

    cy.api('/')

    cy.get('[data-cy="responseBody"]', { timeout: 50 })
      .should('be.visible')
      .and('have.length', currentRetry) // will be 0 on first try, 1 on second

  });

});